File: Programmers/Level2/221020/JavsScript/JadenCase.test.js
import {
  convertToLowerCase, firstWordToUpperCase, solution, toArray,
} from './JadenCase';

test('toLowerCase', () => {
  expect(convertToLowerCase('FoR The lASt weEK')).toBe('for the last week');
});

test('toArray', () => {
  expect(toArray('for the ')).toStrictEqual(['f', 'o', 'r', ' ', 't', 'h', 'e', ' ']);
  expect(toArray(' 3people  Un '))

    .toStrictEqual([' ', '3', 'p', 'e', 'o', 'p', 'l', 'e', ' ', ' ', 'U', 'n', ' ']);
});

test('firstWordToUpperCase', () => {
  expect(firstWordToUpperCase(['f', 'o', 'r', ' ', 't', 'h', 'e', ' ']))
    .toBe('For The ');
});

test('solution', () => {
  expect(solution('for the last week')).toBe('For The Last Week');
  expect(solution('for  the  last  week ')).toBe('For  The  Last  Week ');
});

File: Programmers/Level2/221019/Java/JadenCaseTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class JadenCaseTest {
  @Test
  void split() {
    JadenCase jadenCase = new JadenCase();

    assertArrayEquals(new String[]{"for", "the", "last", "week"},
        jadenCase.split("for the last week"));
  }

  @Test
  void convertToUppearCase() {
    JadenCase jadenCase = new JadenCase();

    String[] sentences = new String[]{"for", "the", "last", "week"};

    assertEquals("For The Last Week", jadenCase.convert(sentences, "for the last week"));
  }

  @Test
  void convertToLowerCase() {
    JadenCase jadenCase = new JadenCase();

    String[] sentences = new String[]{"fOr", "tHE", "lAST", "week"};

    assertEquals("For The Last Week", jadenCase.convert(sentences, "fOr tHE lAST week"));
  }
  @Test
  void solution() {
    JadenCase jadenCase = new JadenCase();

    assertEquals("A Aa " ,jadenCase.solution("a aa "));
  }
}
